cpp/log/etcd_consistent_store-inl.h

```diff
@@ -648,32 +626,38 @@ util::Status EtcdConsistentStore<Logged>::CleanupOldEntries() {
   LOG(INFO) << "Cleaning old entries up to and including sequence number: "
             << clean_up_to_sequence_number;
 
-  std::vector<EntryHandle<Logged>> sequenced_entries;
-  util::Status status(GetSequencedEntries(&sequenced_entries));
+  EntryHandle<ct::SequenceMapping> sequence_mapping;
+  util::Status status(GetSequenceMapping(&sequence_mapping));
   if (!status.ok()) {
-    LOG(WARNING) << "Couldn't get sequenced entries: " << status;
+    LOG(WARNING) << "Couldn't get sequence mapping: " << status;
     return status;
   }
 
-  for (auto& entry : sequenced_entries) {
-    const uint64_t sequence_number(entry.Entry().sequence_number());
-
-    // return if we've hit the limit
-    if (entry.Entry().sequence_number() > clean_up_to_sequence_number) {
-      return util::Status::OK;
-    }
+  size_t i;
+  for (i = 0; i < sequence_mapping.Entry().mapping_size() &&
+                  sequence_mapping.Entry().mapping(i).sequence_number() <=
+                      clean_up_to_sequence_number;
+       ++i) {
+    const uint64_t sequence_number(
+        sequence_mapping.Entry().mapping(i).sequence_number());
 
-    // Otherwise, first we delete the corresponding unsequenced entry
+    // First we delete the corresponding unsequenced entry
     EntryHandle<LoggedCertificate> unseq_entry;
-    const std::string hash(entry.Entry().Hash());
+    const std::string hash(
+        sequence_mapping.Entry().mapping(i).unsequenced_hash());
     status = GetPendingEntryForHash(hash, &unseq_entry);
     if (!status.ok()) {
-      // Log a warning here, but don't bail on the clean up as we could just be
-      // recovering from a crash halfway through a previous clean up.
-      LOG(WARNING) << "Cleanup couldn't get unsequenced entry ("
-                   << util::ToBase64(hash)
-                   << ") corresponding to sequenced entry #" << sequence_number
-                   << " : " << status;
+      if (status.CanonicalCode() == util::error::NOT_FOUND) {
+        // Log a warning here, but don't bail on the clean up as we could just
+        // be recovering from a crash halfway through a previous clean up.
+        LOG(WARNING) << "Cleanup couldn't get unsequenced entry ("
```

Comment:
Yeah, I didn't do it because I thought it might well ripple out, but it's got to be done at some point anyway, may as well be now.
Done.